### PR TITLE
Fix printing on MinGW.

### PR DESCRIPTION
**Pull request by sas**

### CODE REVIEW

File: Headers/DebugServer2/Utils/CompilerSupport.h [hunk 1]
```diff
@@ -33,8 +33,16 @@
 #endif
 
 #if __has_attribute(format)
+#if defined(PLATFORM_MINGW)
+// MinGW uses printf wrappers to provide standard format string behavior on
+// Windows. We need to use __MINGW_PRINTF_FORMAT as GCC assumes MS style printf
+// arguments with `__format__(printf, ...)`.
+#define DS2_ATTRIBUTE_PRINTF(FORMAT, START)                                    \
+  __attribute__((__format__(__MINGW_PRINTF_FORMAT, FORMAT, START)))
+#else
```
Comment (strager) [inline Headers/DebugServer2/Utils/CompilerSupport.h]: Is `__MINGW_PRINTF_FORMAT` a macro? Could we check `#if defined(__MINGW_PRINTF_FORMAT)`?

**Comment by strager** [comment]
lgtm, except I'm not sure how the last two commits are related.

**Comment by sas** [comment]
They're not. :D

On Mon, Apr 4, 2016 at 1:54 PM, strager <notifications@github.com> wrote:

> lgtm, except I'm not sure how the last two commits are related.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/facebook/ds2/pull/247#issuecomment-205492787>
>



-- 
Stephane Sezer

File: Headers/DebugServer2/Utils/CompilerSupport.h (hunk 1 above)
Comment (sas) [inline Headers/DebugServer2/Utils/CompilerSupport.h]: Meh. It makes a bit more sense to me this way. __MINGW_PRINTF_FORMAT is not a "feature", it's how mingw is supposed to be used, as I understand it.

**Comment by fjricci** [comment]
@facebook-github-bot shipit

**Comment by sas** [comment]
Committed as 42c31115f3fa50c5c4fd952e24d37b4e30704f71..15edd068ba656603f7d2697a27b0738ad9c8df00.

**Pull request closed without merging**